### ISSUE TITLE
Add support for deploying Forms

### ISSUE THREAD

**Issue by jwulf** [camunda-community-hub/zeebe-client-node-js#332]
This relates to an epic to allow [Form deployment and binding](https://github.com/camunda/product-hub/issues/419).

The Zeebe epic is here: [[EPIC] Deploy Forms to Zeebe](https://github.com/camunda/zeebe/issues/12874).

The Zeebe gRPC API will allow a form to be deployed via DeployResources.

The relevant gRPC API change is proposed as:

```grpc
message DeployResourceResponse {
 // the unique key identifying the deployment
 int64 key = 1;
 // a list of deployed resources, e.g. forms or processes
 repeated Deployment deployments = 2;
}

message Deployment {
 // each deployment has only one metadata
 oneof Metadata {
   // metadata of a deployed process
   ProcessMetadata process = 1;
   // metadata of a deployed decision
   DecisionMetadata decision = 2;
   // metadata of a deployed decision requirements
   DecisionRequirementsMetadata decisionRequirements = 3;
   // metadata of a deployed form
   FormMetadata form = 4;
 }

message FormMetadata {
 // the form ID, as parsed during deployment; together with the
 // versions forms a unique identifier for a specific form
 string formId = 1;
 // the assigned form version
 int32 version = 2;
 // the assigned key, which acts as a unique identifier for this form
 int64 formKey = 3;
 // the resource name
 string resourceName = 4;
}
```

**Comment by jwulf** [comment]
At this point, I can't find a branch with the new gRPC API - it looks like it's still too early.

**Comment by jwulf** [comment]
It is now baked in - it's another method signature for `deployResource`. Needs to be tested against both Camunda SaaS and Self-Managed.

**Comment by jwulf** [comment]
This is [slated for 8.4-alpha.2](https://github.com/camunda/product-hub/issues/419#issuecomment-1774709777). 

The feature is in a branch and ready to test against an alpha release.

**Comment by jwulf** [comment]
Published in 8.3.1

**Comment by philippfromme** [comment]
Nice! 🚀
